### PR TITLE
Added packages for the Rust compiler and Cargo

### PR DESCRIPTION
**Pull request by esteve**
Please add the following dependency to the rosdep database.

## Package name:

cargo
rustc

## Package Upstream Source:

https://github.com/rust-lang/cargo
https://github.com/rust-lang/rust

## Purpose of using this:

These two packages are being used by colcon-cargo, colcon-ros-cargo, rclrs and rclrs_common

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/search?keywords=cargo
  - https://packages.debian.org/search?keywords=rustc
- Ubuntu: https://packages.ubuntu.com/
  - https://packages.ubuntu.com/search?keywords=cargo
  - https://packages.ubuntu.com/search?keywords=rustc

### CODE REVIEW

**Comment by sloretz** [comment]
@esteve on the files changed tab there are a couple annotations that the keys could be resolvable on fedora and rhel. Mind adding the fedora and rhel entries?

**Comment by esteve** [comment]
@sloretz sorry for the slow response, I've added the keys for Fedora and RHEL in https://github.com/ros/rosdistro/commit/3a79fe5ebbfeb6a04fb134a340414d5f37bf3335

File: rosdep/base.yaml [hunk 1]
```diff
@@ -448,6 +448,11 @@ can-utils:
   fedora: [can-utils]
   nixos: [can-utils]
   ubuntu: [can-utils]
+cargo:
+  debian: [cargo]
+  fedora: [cargo]
```
Comment (nuclearsandwich) [inline rosdep/base.yaml:453]:
```suggestion
  fedora: [cargo]
  opensuse: [cargo]
```

File: rosdep/base.yaml [hunk 2]
```diff
@@ -6739,6 +6744,11 @@ rtmidi:
   fedora: [rtmidi-devel]
   nixos: [rtmidi]
   ubuntu: [librtmidi-dev]
+rustc:
```
Comment (nuclearsandwich) [inline rosdep/base.yaml:6747]:
```suggestion
rustc:
  archlinux: [rust]
```

File: rosdep/base.yaml [hunk 3]
```diff
@@ -6739,6 +6744,11 @@ rtmidi:
...
+  debian: [rustc]
+  fedora: [rust]
```
Comment (nuclearsandwich) [inline rosdep/base.yaml:6749]:
```suggestion
  fedora: [rust]
  opensuse: [rust]
```

**Review by nuclearsandwich — APPROVED** [review]
Added suggestions for Archlinux and OpenSUSE but looks good otherwise.

**Comment by esteve** [comment]
@nuclearsandwich I've applied your suggestions, thanks! Is this now ready to merge?

**Review by wjwwood — APPROVED** [review]
lgtm, @nuclearsandwich can I merge?

**Review by tfoote — APPROVED** [review]
lgtm

**Pull request merged**